src/lib/types.ts

```diff
@@ -92,13 +92,23 @@ export interface UserAttributes {
    * An email change token.
    */
   email_change_token?: string
-
   /**
-   * A custom data object. Can be any JSON.
+   * A custom data object for user_metadata that a user can modify. Can be any JSON.
    */
   data?: object
 }
 
+export interface AdminUserAttributes extends UserAttributes {
+  /**
+   * A custom data object for app_metadata that only a service role can modify.
+   *
+   * Is JSON that includes app-specific info, such as identity providers, roles, and other
+   * access control information.
+   */
```

Comment:
should we also add the boolean fields `emailConfirm` and `phoneConfirm` here? 

see https://github.com/supabase/gotrue/blob/master/api/admin.go#L21-L24

Reply:
Ah, yes, we should so that the types match up with GoTrue's:

```
type adminUserParams struct {
	Aud          string                 `json:"aud"`
	Role         string                 `json:"role"`
	Email        string                 `json:"email"`
	Phone        string                 `json:"phone"`
	Password     string                 `json:"password"`
	EmailConfirm bool                   `json:"email_confirm"`
	PhoneConfirm bool                   `json:"phone_confirm"`
	UserMetaData map[string]interface{} `json:"user_metadata"`
	AppMetaData  map[string]interface{} `json:"app_metadata"`
}
```